packages/amplify-provider-awscloudformation/src/utils/upload-auth-trigger-template.ts

```diff
@@ -0,0 +1,40 @@
+import { $TSContext, $TSObject, FeatureFlags, JSONUtilities, pathManager } from 'amplify-cli-core';
+import { S3 } from '../aws-utils/aws-s3';
+import path from 'path';
+import fs from 'fs-extra';
+import * as constants from '../constants';
+
+export const AUTH_TRIGGER_TEMPLATE = 'auth-trigger-cloudformation-template.json';
+export const AUTH_TRIGGER_STACK = 'AuthTriggerCustomLambdaStack';
+export async function uploadAuthTriggerTemplate(context: $TSContext) {
+  if (!FeatureFlags.getBoolean('auth.breakCircularDependency')) {
+    return {};
+  }
+  const category = 'auth';
+  let { amplifyMeta } = context.amplify.getProjectDetails();
+  const authResource = amplifyMeta?.auth ?? {};
+  const authTriggerCfnFilePath = path.join(pathManager.getBackendDirPath(), category, Object.keys(authResource)[0], AUTH_TRIGGER_TEMPLATE);
+  if (fs.existsSync(authTriggerCfnFilePath)) {
+    const cfnObject: $TSObject = JSONUtilities.readJson(authTriggerCfnFilePath);
+    const S3_UPLOAD_PATH = path.join(category, AUTH_TRIGGER_TEMPLATE);
+
+    return {
+      AuthTriggerTemplateURL: await uploadCfnToS3(context, S3_UPLOAD_PATH, cfnObject),
+    };
+  } else {
+    return {
+      AuthTriggerTemplateURL: '',
```

Comment:
Is there any benefit to returning this with the empty string instead of returning `{}` like on line 11? I think returning `{}` would simplify the logic in `packages/amplify-provider-awscloudformation/src/push-resources.ts` as well. Right now, that logic looks for the empty string, but sometimes that value can be `undefined`.